src/sql/connection.py

```diff
@@ -195,11 +195,19 @@ def _error_module_not_found(cls, e):
         return ModuleNotFoundError("test")
 
     def __init__(self, engine, alias=None):
-        self.dialect = engine.url.get_dialect()
-        self.metadata = sqlalchemy.MetaData(bind=engine)
+        self.url = engine.url
+        self.dialect = self.url.get_dialect()
+
+        version = int(sqlalchemy.__version__.split(".")[0])
+        if version < 2:
+            self.metadata = sqlalchemy.MetaData(bind=engine)
+
         self.name = self.assign_name(engine)
-        self.session = engine.connect()
-        self.connections[alias or repr(self.metadata.bind.url)] = self
+        self.internal_connection = engine.connect()
```

Comment:
looks like this name was chosen arbitrarily in the original PR, was there any mention on why such rename? if no, we can keep the original name since a rename will impact other open PRs

Reply:
I reverted the name to `session`.